fetch post-GA corrections and additions (#18610)
cockroachdb/docs: src/current/v23.2/molt-fetch.md

- If you plan to use continuous replication (using either [`--ongoing-replication`](#replication) or the [CDC cursor](#cdc-cursor)):

> 	- The source and target database names **must** be identical when using `--ongoing-replication`.

Reviewer: Are those changes merged in and released in the latest `replicator` binary?

> 	- The source and target database names **must** be identical when using `--ongoing-replication`.

Reviewer 2: The latest version of replicator allows you to have a different database name for the target.

> 	- The source and target database names **must** be identical when using `--ongoing-replication`.

Reviewer: @taroface , which means we can probably remove this requirement and doc now.

| `--crdb-pts-refresh-interval`                 | The frequency at which the protected timestamp's validity is extended. This interval maintains protection of the data snapshot until data export from a CockroachDB source is completed. For example, if set to `10m`, the protected timestamp's expiration will be extended by the duration specified in `--crdb-pts-duration` (e.g., `24h`) every 10 minutes while export is not complete. <br><br>**Default:** `10m0s`                                                                                                                      |
| `--direct-copy`                               | Enables [direct copy mode](#fetch-mode), which copies data directly from source to target without using an intermediate store.                                                                                                                                                                                                                                                                                                                                                                                                                 |

> | `--export-concurrency`                        | Number of shards to export at a time, each on a dedicated thread. This only applies when exporting data from the source database, not when loading data into the target database. **Note:** The number of concurrent threads is the product of `--export-concurrency` and `--table-concurrency`. See [Best practices](#best-practices).<br><br>This value **cannot** be set higher than `1` when moving data from MySQL. Refer to [Best practices](#best-practices).<br><br>**Default:** `4` with a PostgreSQL source; `1` with a MySQL source |

Reviewer 3: Do you need two links to the Best Practices section in such close proximity?

Author: Oops! Good catch

| `--flush-rows`                                | Number of rows before the source data is flushed to intermediate files. **Note:** If `--flush-size` is also specified, the fetch behavior is based on the flag whose criterion is met first.                                                                                                                                                                                                                                                                                                                                                   |
| `--flush-size`                                | Size (in bytes) before the source data is flushed to intermediate files. **Note:** If `--flush-rows` is also specified, the fetch behavior is based on the flag whose criterion is met first.                                                                                                                                                                                                                                                                                                                                                  |

> | `--import-batch-size`                         | The number of files to be imported at a time to the target database. This applies only when using the [`IMPORT INTO` mode](#fetch-mode) for loading data into the target. **Note:** Increasing this value can improve the performance of full-scan queries on the target database shortly after fetch completes, but very high values are not recommended. If any one file in the import batch fails, you will need to [retry](#fetch-continuation) the entire batch.<br><br>**Default:** `1000`                                               |

Reviewer 3: ```suggestion
| `--import-batch-size`                         | The number of files to be imported at a time to the target database. This applies only when using the [`IMPORT INTO` mode](#fetch-mode) for loading data into the target. **Note:** Increasing this value can improve the performance of full-scan queries on the target database shortly after a fetch completes, but very high values are not recommended. If any individual file in the import batch fails, you must [retry](#fetch-continuation) the entire batch.<br><br>**Default:** `1000`                                               |
```

| `--local-path`                                | The path within the [local file server](#local-file-server) where intermediate files are written (e.g., `data/migration/cockroach`). `--local-path-listen-addr` must be specified.                                                                                                                                                                                                                                                                                                                                                             |

> | `--local-path-crdb-access-addr`               | Address of a [local file server](#local-file-server) that is reachable by CockroachDB. This flag is only necessary if CockroachDB cannot reach the local address specified with `--local-path-listen-addr` (e.g., when moving data to a CockroachDB {{ site.data.products.cloud }} deployment).<br><br>**Default:** Value of `--local-path-listen-addr`. `--local-path` and `--local-path-listen-addr` must be specified.                                                                                                                      |

Reviewer 3: I don't really understand what the Default bit means.

| `--flush-rows`                                | Number of rows before the source data is flushed to intermediate files. **Note:** If `--flush-size` is also specified, the fetch behavior is based on the flag whose criterion is met first.                                                                                                                                                                                                                                                                                                                                                   |
| `--flush-size`                                | Size (in bytes) before the source data is flushed to intermediate files. **Note:** If `--flush-rows` is also specified, the fetch behavior is based on the flag whose criterion is met first.                                                                                                                                                                                                                                                                                                                                                  |

> | `--import-batch-size`                         | The number of files to be imported at a time to the target database. This applies only when using the [`IMPORT INTO` mode](#fetch-mode) for loading data into the target. **Note:** Increasing this value can improve fetch performance, but very high values are not recommended. If any one file in the import batch fails, you will need to [retry](#fetch-continuation) the entire batch.<br><br>**Default:** `10`                                                                                                                         |

Reviewer: Default is actually `1000` now.

Reviewer: @ZhouXing19 to add her review here.

> | `--import-batch-size`                         | The number of files to be imported at a time to the target database. This applies only when using the [`IMPORT INTO` mode](#fetch-mode) for loading data into the target. **Note:** Increasing this value can improve fetch performance, but very high values are not recommended. If any one file in the import batch fails, you will need to [retry](#fetch-continuation) the entire batch.<br><br>**Default:** `10`                                                                                                                         |

Reviewer 4: ```suggestion
| `--import-batch-size`                         | The number of files to be imported at a time to the target database. This applies only when using the [`IMPORT INTO` mode](#fetch-mode) for loading data into the target. **Note:** Increasing this value can improve the performance of full-scan queries on the target database shortly after the fetch process finishes, but very high values are not recommended. If any one file in the import batch fails, you will need to [retry](#fetch-continuation) the entire batch.<br><br>**Default:** `10`                                                                                                                         |
```

--schema-filter 'public'
~~~

> `--table-filter` specifies a range of tables to move to CockroachDB, based on a POSIX regex string. For example, to move every table in the source database that has "user" in the title:

Reviewer 3: ```suggestion
`--table-filter` specifies a range of tables to move to CockroachDB,formatted as a POSIX regex string. For example, to move every table in the source database that has "user" in the title:
```

--schema-filter 'public'
~~~

`--table-filter` specifies a range of tables to move to CockroachDB, based on a POSIX regex string. For example, to move every table in the source database that has "user" in the title:

{% include_cached copy-clipboard.html %}
~~~

> --table-filter '.*user.*'

Reviewer: Awesome, thanks for fixing this example!

Before using this feature, complete the following:

> - Ensure that the source and target database names are **identical**. Otherwise, replication will not work.

Reviewer: No longer needed (reference Silvano's comment at the top)
